Update styleguide colors and incorporate new style guide created by @cschneebeck

caseycesari opened the pull request:
This PR does a few things related to the `styleguide` app, and the framework itself:

- The framework DOM and stylesheet used in the `styleguide` were out-of-date with the app. They are updated to the latest here.
- @cschneebeck's style guide was added to the `styleguide` app and is accessible at `http://localhost:3000/snippet_index.html` when running the `styleguide` dev server.
- Components added in the new style guide were added to the existing style guide.
- Some of the base colors used in `styleguide` were updated per #875.
- The framework was updated with the latest version of the `styleguide` app stylesheet and the new CSS and JS dependencies needed by the new components added in @cschneebeck's style guide.

*Existing style guide, new components, link to new style guide*
![image](https://cloud.githubusercontent.com/assets/1042475/23186627/d1283e14-f855-11e6-8bab-3e258aefc17b.png)

*New style guide, with link to existing style guide*
![image](https://cloud.githubusercontent.com/assets/1042475/23186685/ee7d4496-f855-11e6-9d30-d2d1e85c6938.png)


**Testing**
- Run the `styleguide` app and verify that 1) the existing styleguide has the new components, 2) the styleguides are linked, and 3) the new styleguide works.
- Verify the color changes in 21c8345 satisfy #875 and are effective.
- Verify that the framework has been updated with the latest stylesheets and dependencies.

Connects to #883 
Connects to #875

mmcfarland commented on styleguide/app/index.html:
```diff
@@ -196,6 +207,78 @@ <h2 class="sidebar-content-heading">Components</h2>
                         </section>
+                        <section>
+                            <h3>Range slider</h3>
+                            <div>
```
I think this element needs `class="slider-container"` to get teh handles to light up correctly.

![screenshot from 2017-02-21 17 39 17](https://cloud.githubusercontent.com/assets/1014341/23188433/aeda55de-f85c-11e6-82ad-9781c406f287.png)

caseycesari replied:
Fixed in bc771cc

mmcfarland commented on styleguide/app/index.html:
```diff
@@ -196,6 +207,78 @@ <h2 class="sidebar-content-heading">Components</h2>
                                 <div class="slider"></div>
```
This "old" slider now has a permanently enabled tooltip that doesn't exist on the current version

![screenshot from 2017-02-21 17 39 49](https://cloud.githubusercontent.com/assets/1014341/23188479/d421d11e-f85c-11e6-9e00-5fa16b1ffdf0.png)

caseycesari replied:
Fixed in bc771cc

mmcfarland commented on styleguide/app/index.html:
```diff
@@ -19,63 +21,72 @@
             </ul>
         </div>
+        <div class="nav-region-subtitle">
+            <a href="/snippet_index.html">View "code snippet" style guide</a>
```
Consider making this `color: #fff` so that it contrasts a bit more.
![screenshot from 2017-02-21 17 42 23](https://cloud.githubusercontent.com/assets/1014341/23188553/208ea0d6-f85d-11e6-8c78-6f7f200c87be.png)

caseycesari replied:
Fixed in d0952b4

caseycesari commented:
Ready for another look.

mmcfarland commented:
Got a lot closer - all my previous issues have been fixed, nice job.

I am noticing two styling issues with the new components (accordian and chosen), which don't have the corresponding custom styles applied to them in the style guide app (but do on http://cschneebeck.com/styleguide/).   I think the `styleguide.css` needs the appropriate styles from `newComponents.css` to be included to get those specific overrides.

Note the border-radius and the flat box.  

![screenshot from 2017-02-22 16 22 52](https://cloud.githubusercontent.com/assets/1014341/23233023/76d9c998-f91b-11e6-8a11-9569910cf4d2.png)
![screenshot from 2017-02-22 16 20 44](https://cloud.githubusercontent.com/assets/1014341/23233024/76dbd3c8-f91b-11e6-8db4-6984d599a12e.png)

This is how they're intended to look:
![screenshot from 2017-02-22 16 27 37](https://cloud.githubusercontent.com/assets/1014341/23233119/d8007906-f91b-11e6-8ece-308ff5289705.png)
![screenshot from 2017-02-22 16 27 21](https://cloud.githubusercontent.com/assets/1014341/23233118/d7fee4f6-f91b-11e6-8de4-c0858803380a.png)

caseycesari commented:
Thanks. Pushed a fixup commit. The chosen style problem was the result of a typo. The accordion problem was fixed by including `newComponents.css` in the styleguide `index.html`. These rules were already included in the styleguide `main.css`, but I think the problem has to do with rule ordering/precedent. If I moved the `main.css` to be the last stylesheet to include, that fixed the accordian, but made the slider perma-tooltip comeback. In any case, both issues should be fixed now.

caseycesari commented:
@cschneebeck I added the new section ("Class Names and Custom Styles") as well as updated `style.css` used in your styleguide.

caseycesari commented:
@cschneebeck You can see the latest version of your style guide here: http://coastalresiliencenetwork.github.io/GeositeFramework/snippet_index.html

Ignore the styleguide at http://coastalresiliencenetwork.github.io/GeositeFramework/ for now. It's got some broken links at the moment.

caseycesari commented:
I updated `initialize.js` on http://coastalresiliencenetwork.github.io/GeositeFramework/snippet_index.html, which fixed the rounded accordion corners and the slider width.

I also fixed the assets for http://coastalresiliencenetwork.github.io/GeositeFramework/. That should be in good shape now too.

caseycesari commented:
Okay, everything should be up-to-date.

cschneebeck commented:
Awesome! Everything looks good. Thanks for all your help!

Cheers,
Casey Schneebeck
Web Developer | The Nature Conservancy
North America Science | Boulder, CO
415-722-9768

From: Casey Cesari <notifications@github.com>
Reply-To: CoastalResilienceNetwork/GeositeFramework <reply@reply.github.com>
Date: Thursday, March 2, 2017 at 9:04 AM
To: CoastalResilienceNetwork/GeositeFramework <GeositeFramework@noreply.github.com>
Cc: Casey Schneebeck <cschneebeck@TNC.ORG>, Mention <mention@noreply.github.com>
Subject: Re: [CoastalResilienceNetwork/GeositeFramework] Update styleguide colors and incorporate new style guide created by @cschneebeck (#893)


Okay, everything should be up-to-date.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/CoastalResilienceNetwork/GeositeFramework/pull/893#issuecomment-283695603>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AK6qwppC0SUIpidPJAALX7QYbcSPIpVWks5rhuiUgaJpZM4MH6d5>.

caseycesari commented:
Thanks. Merging now.
